Multi protocol updates to MPM version 1.2.1.71

MRC3742 opened the pull request:
Multiprotocol listing updates for Companion GUI and a few changes, updates to Radio GUI.
Updates to bring OpenTX version 2.3 Multi listings in line to current MPM version 1.2.1.71

Adds new Devo Sub Protocols to Companion and Radio, which updates to current version.
Various fixes and changes to make Protocols more understandable and easier to locate.

@pascallanger Please Review
As I added the Fixed ID Value to Devo Protocol as was added in the Jumper branch.
I believe you were waiting for my earlier commit of 1.2.1.63 to be merged before you added it here?

Compiled, tested and working using OpenTX Companion v2.3

MRC3742 committed:
Multi protocol updates to MPM version 1.2.1.71

3djc commented on radio/src/translations/untranslated.h:
```diff
@@ -129,5 +129,5 @@
-#define LEN_MULTI_PROTOCOLS            "\007"
-#define TR_MULTI_PROTOCOLS             "FlySky\0""Hubsan\0""FrSky\0 ""Hisky\0 ""V2x2\0  ""DSM\0   ""Devo\0  ""YD717\0 ""KN\0    ""SymaX\0 ""SLT\0   ""CX10\0  ""CG023\0 ""Bayang\0""ESky\0  ""MT99XX\0""MJXq\0  ""Shenqi\0""FY326\0 ""SFHSS\0 ""J6 PRO\0""FQ777\0 ""Assan\0 ""Hontai\0""OpenLrs""FS 2A\0 ""Q2x2\0  ""Walk.\0 ""Q303\0  ""GW008\0 ""DM002\0 ""Cabell\0""Esy150\0""H8 3D\0 ""Corona\0""CFlie\0 "" ...
+#define LEN_MULTI_PROTOCOLS            "\006"
+#define TR_MULTI_PROTOCOLS             "FlySky""Hubsan""FrSky\0""Hisky\0""V2x2\0 ""DSM\0  ""Devo\0 ""YD717\0""KN\0   ""SymaX\0""SLT\0  ""CX10\0 ""CG023\0""Bayang""ESky\0 ""MT99XX""MJXq\0 ""Shenqi""FY326\0""SFHSS\0""J6 PRO""FQ777\0""Assan\0""Hontai""OpnLrs""FS 2A\0""Q2x2\0 ""Walk.\0""Q303\0 ""GW008\0""DM002\0""Cabell""Esy150""H8 3D\0""Corona""CFlie\0""Hitec\0""WFly\0 ""Bugs\0 ""BugMin""Traxas""NC1 ...
```
I’m not too fan of having shortened names on radio where it matters more, and different names in companion. Maybe I should make room on radio to display the ‘full’ name ?

bsongis replied:
7 chars are perfect on 128x64 screens, I don't see why we would shorten to 6 chars

3djc replied:
If we allow more space,we could use the same list as companion : https://github.com/opentx/opentx/blob/2877077fab6320eae25f2d7ce17d3c298690a0f2/companion/src/firmwares/multiprotocols.cpp#L156

raphaelcoeffic replied:
Using the same table for the options (`multi_protocols` & `multiProtocols`) would be a good idea as well, so they don't have to be kept in sync. It seems already prepared, so why not doing it?

MRC3742 replied:
@3djc The names on the Radio have always been six characters in length.
PR #6641 changed it to seven that was merged earlier today.

Currently on the  "212 x 64" screen all items are listed on one line.
eg. Mode  MULT FS_2A  PWM,IBUS

4 Spaces are allotted for module type
6 Spaces are allotted for Protocol
8 Spaces can be used for Sub Protocol with the last character partially hidden by running off screen
![IMG_3243](https://user-images.githubusercontent.com/26642502/63308537-16e39100-c2c1-11e9-8d7a-0caa81645490.jpg)

In the narrower screen, (128x64) as on the QX7 or X-Lite, the Subtype is shown on a separate line

I am just trying to work within the boundaries that are currently available?
If more space is allotted then we could definitely have more descriptive names.

raphaelcoeffic replied:
> Currently on the 128 x 64 screen all items are listed on one line. eg. Mode MULT FS_2A PWM,IBUS

This is on 2.3 (X9 lite):

![IMG_6526](https://user-images.githubusercontent.com/1050031/63305094-d34f4e00-c2e5-11e9-8fde-678821404b56.jpg)

MRC3742 replied:
My mistake
I meant the 212 x 64 screen everything is on one line. (now edited to correct)

The 128 x 64 is the narrower screen I refer to above of the QX7, X-Lite (and X9 lite)
    Thanks for the correction and screenshot!

raphaelcoeffic reviewed (commented):
Maybe not related to this PR, but we should really use the same structs/enums in both the firmware and companion as much as possible. Feels quite tedious and dangerous to have keep these in sync.

bsongis commented:
![image](https://user-images.githubusercontent.com/5603504/63321869-4d052d00-c322-11e9-8100-689ba85d0e7f.png)

pascallanger commented:
If you ever go to 7 caracters then you need to expand the protocol names:
`#define TR_MULTI_PROTOCOLS             "FlySky\0""Hubsan\0""FrSky\0 ""Hisky\0 ""V2x2\0  ""DSM\0   ""Devo\0  ""YD717\0 ""KN\0    ""SymaX\0 ""SLT\0   ""CX10\0  ""CG023\0 ""Bayang\0""ESky\0  ""MT99XX\0""MJXq\0  ""Shenqi\0""FY326\0 ""SFHSS\0 ""J6 Pro\0""FQ777\0 ""Assan\0 ""Hontai\0""OpenLrs""FSky 2A""Q2x2\0  ""Walkera""Q303\0  ""GW008\0 ""DM002\0 ""Cabell\0""Esky150""H8 3D\0 ""Corona\0""CFlie\0 ""Hitec\0 ""WFly\0  ""Bugs\0  ""BugMini""Traxas\0""NCC1701""E01X\0  ""V911S\0 ""GD00X\0 ""V761\0  ""KF606\0 ""Redpine""Potensi""ZSX\0   ""Flyzone"`
Going to 7 caracters, you only have 2 names that are truncated: Bug**s**Mini and Potensi**c**

3djc commented:
I would be thinking more to have the full name on one line (on 128) to allow the same names as companion

bsongis commented:
@pascallanger when you have such strings, like "FlySky " (with a space at the end) it's not nice when the field is inverted or blinking. Just use this "FlySky\0" is better

bsongis commented:
And yes 7 chars is perfect. I wonder if 8 would fit. We use a non monospace font, so perhaps yes.

pascallanger commented:
Agreed I've fixed it already on my pull request to JumperTX:
`#define LEN_MULTI_PROTOCOLS            "\007"
#define TR_MULTI_PROTOCOLS             "FlySky\0""Hubsan\0""FrSky\0 ""Hisky\0 ""V2x2\0  ""DSM\0   ""Devo\0  ""YD717\0 ""KN\0    ""SymaX\0 ""SLT\0   ""CX10\0  ""CG023\0 ""Bayang\0""ESky\0  ""MT99XX\0""MJXq\0  ""Shenqi\0""FY326\0 ""SFHSS\0 ""J6 Pro\0""FQ777\0 ""Assan\0 ""Hontai\0""OpenLrs""FSky 2A""Q2x2\0  ""Walkera""Q303\0  ""GW008\0 ""DM002\0 ""Cabell\0""Esky150""H8 3D\0 ""Corona\0""CFlie\0 ""Hitec\0 ""WFly\0  ""Bugs\0  ""BugMini""Traxas\0""NCC1701""E01X\0  ""V911S\0 ""GD00X\0 ""V761\0  ""KF606\0 ""Redpine""Potensi""ZSX\0   ""Flyzone"`
I've also added in my JumperTX pull request:
- Changed string length of modules from 4 to 5 to allow "MULTI" to appear fully instead of "MULT":
`#define LEN_TARANIS_PROTOCOLS  "\005"
#define TR_TARANIS_PROTOCOLS   "OFF\0 ""PPM\0 ""XJT\0 ""DSM2\0""CRSF\0""MULTI""R9M\0 ""SBUS\0"`
 - Changed "Bind on powerup" to "Bind on channel" which is the default module configuration.
`#define TR_MULTI_AUTOBIND      TR(INDENT "Autobind",INDENT "Bind on channel")`

pascallanger commented:
I know I'm talking about the JumperTX fork here but I'm trying to propose the modifications in both places... and I have a T16 so I'm trying the changes there first...

bsongis commented:
Yes, "MULTI" instead of "MULT" is far better.

MRC3742 committed:
Merge branch '2.3' of https://github.com/opentx/opentx into 2.3

MRC3742 committed:
Merge branch '2.3' of https://github.com/opentx/opentx into 2.3

MRC3742 committed:
Update Multi Protocol descriptions

MRC3742 commented:
Changed Module string from MULT to MULTI.
Reverted Protocol string from 6 to 7 characters with changes allowing more complete descriptions.
Changed TR_MULTI_AUTOBIND from "Bind on powerup" to "Bind on channel" as per pascallanger changes in Jumper fork.

pascallanger commented:
Great, it will look far more better.

3djc commented:
Thanks all
